runtime/major_gc.c

```diff
@@ -401,8 +410,8 @@ static void init_sweep_phase(void)
   if (caml_major_gc_hook) (*caml_major_gc_hook)();
 }
 
-/* auxiliary function of mark_slice */
-Caml_inline void mark_slice_darken(struct mark_stack* stk, value v, mlsize_t i,
+/* auxiliary function of mark_ephe_aux */
+Caml_inline void mark_ephe_darken(struct mark_stack* stk, value v, mlsize_t i,
                                        int in_ephemeron, int *slice_pointers,
                                        intnat *work)
```

Comment:
I'm not going to change the ephemeron path in this PR, because there are a number of known bugs being worked on in that code, and I don't want to cause conflicts with #9424. Once both are merged, we can do some cleanup.